src/vmm/src/seccomp.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+use std::io::Read;
+use std::sync::Arc;
+
+use bincode::{DefaultOptions, Error as BincodeError, Options};
+
+/// Each BPF instruction is 8 bytes long and 4 byte aligned.
+/// This alignment needs to be satisfied in order for a BPF code to be accepted
+/// by the syscalls. Using u64 here is is safe as it has same size and even bigger alignment.
+pub type BpfInstruction = u64;
+
+/// Program made up of a sequence of BPF instructions.
+pub type BpfProgram = Vec<BpfInstruction>;
+
+/// Reference to program made up of a sequence of BPF instructions.
+pub type BpfProgramRef<'a> = &'a [BpfInstruction];
+
+/// Type that associates a thread category to a BPF program.
+pub type BpfThreadMap = HashMap<String, Arc<BpfProgram>>;
+
+/// Retrieve empty seccomp filters.
+pub fn get_empty_filters() -> BpfThreadMap {
+    let mut map = BpfThreadMap::new();
+    map.insert("vmm".to_string(), Arc::new(vec![]));
+    map.insert("api".to_string(), Arc::new(vec![]));
+    map.insert("vcpu".to_string(), Arc::new(vec![]));
+    map
+}
+
+/// Binary filter deserialization errors.
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum DeserializationError {
+    /// Bincode deserialization failed: {0}
+    Bincode(BincodeError),
+}
```

Comment:
nit: do we really need an enum with just one variant here? Can we just use `BincodeError` directly?

Reply:
Firecracker seccomp.rs needs this because FC does not depend on bincode directly.

Reply:
Ah, annoying. Could solve that by using a public reexport of BincodeError inside this file (e.g. `pub use bincode::BincodeError as DeserializationError`)

Reply:
yup, this work